File: LeetCode/852山脉数组的峰顶索引.go
func peakIndexInMountainArray(arr []int) int {
	left, right := 1, len(arr)-2
	ind := 0
	for left <= right {
		mid := left + (right-left)/2
		if arr[mid] > arr[mid+1] {
			ind = mid
			right = mid - 1
		} else {
			left = mid + 1
		}
	}
	return ind
}

File: LeetCode/852山脉数组的峰顶索引.java
class Solution {
    public int peakIndexInMountainArray(int[] arr) {
        int left = 1, right = arr.length - 2;
        int ind = 0;
        while (left <= right) {
            int mid = left + (right - left) / 2;
            if (arr[mid] > arr[mid + 1]) {
                ind = mid;
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }
        return ind;
    }
}